mediagrains/gsf.py

```diff
@@ -1514,8 +1514,8 @@ def add_grains(self,
     def dump(self):
         """Dump the whole contents of this encoder to the file in one go,
         replacing anything that's already there."""
-        with self:
-            pass
+        self.start_dump(all_at_once=True)
+        self.end_dump(all_at_once=True)
```

Comment:
Could be `self.end_dump()` but that because all_at_once is not used in end_dump(). That could be left to a separate PR if you want.

Reply:
It's not used in this class's implementation, but it was part of the declared interface before, and hence there's no guarantee that subclasses that override `end_dump` won't use it. Hence I've included it.